src/frontend/js/hooks/useDashboardRoutes/index.spec.tsx

```diff
@@ -0,0 +1,71 @@
+import { QueryClientProvider } from 'react-query';
+import { IntlProvider } from 'react-intl';
+import { renderHook } from '@testing-library/react-hooks';
+import { PropsWithChildren } from 'react';
+import * as mockFactories from 'utils/test/factories';
+import useDashboardRoutes from './index';
+import createQueryClient from '../../utils/react-query/createQueryClient';
+import JoanieApiProvider from '../../data/JoanieApiProvider';
+
+jest.mock('utils/context', () => ({
+  __esModule: true,
+  default: mockFactories
+    .ContextFactory({
+      authentication: { backend: 'fonzie', endpoint: 'https://demo.endpoint' },
+      joanie_backend: { endpoint: 'https://joanie.endpoint' },
+    })
+    .generate(),
+}));
+
+jest.mock('react-intl', () => {
+  const originIntl = jest.requireActual('react-intl');
+  return {
+    ...originIntl,
+    useIntl: () => ({
+      formatMessage: (message: string) => {
+        return message;
+      },
+    }),
+  };
+});
+
+jest.mock('components/DashBoard/routes', () => [
+  {
+    element: 'whatever',
+    intlPath: 'home',
+    intlTitle: 'home',
+    protected: false,
+  },
+  {
+    element: '<h2>Whatever</h2>',
+    intlPath: 'protected',
+    intlTitle: 'protected',
+    protected: true,
+  },
+]);
+
+describe('useDashBoardRoutes', () => {
+  const wrapper = ({ children }: PropsWithChildren<{}>) => (
+    <IntlProvider locale="en">
+      <QueryClientProvider client={createQueryClient()}>
+        <JoanieApiProvider>{children}</JoanieApiProvider>
```

Comment:
`<JoanieApiProvider>` is useless I think.
```suggestion
        {children}
```